components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java

```diff
@@ -3165,6 +3185,31 @@ private void updateRegistryResources(Registry registry, String artifactPath, Str
         }
     }
 
+    private void updateRegistryResourcesForArtifacts(Registry registry, String apiId, String artifactResourcePath)
+            throws RegistryException {
+
+        //get path to API in registry
+        String artifactPath = GovernanceUtils.getArtifactPath(registry, apiId);
+
+        //get API
+        Resource apiResource = registry.get(artifactPath);
```

Comment:
In this method, we do couple of registry get actions to get the API's organization. I think we could pass the organization visibility from the method calling location instead of doing multiple registry calls. For example, API artifact is already retrieved in https://github.com/wso2/carbon-apimgt/pull/12717/files#diff-e193f41af36b63c9f226e35c981eeae30ee45850c880876c502785b547dbe00cR2635. you could get the organization information there and pass it to this method

Reply:
Fixed.